src/backend/joanie/payment/models.py

```diff
@@ -369,6 +370,32 @@ def get_cards_for_owner(self, username):
             owners__username=username, payment_provider=payment_provider.name
         )
 
+    def delete_unused(self):
+        """
+        Delete all unused credit cards.
+        """
+        unlinked_credit_cards = []
+        deleted_credit_cards = []
+
+        no_card_order_states = enums.ORDER_INACTIVE_STATES + (
+            enums.ORDER_STATE_COMPLETED,
+        )
+
+        for credit_card in self.all():
+            for order in credit_card.orders.all():
+                if order.state in no_card_order_states:
```

Comment:
Do you really need to iterate over all orders ?
```suggestion
             for order in credit_card.orders.filter(state__in=no_card_order_states):
```